### PR TITLE
Add option to dump profiler stats

### PR DESCRIPTION
**Pull request by Dosenpfand**
Add a config option `DEBUG_TB_PROFILER_DUMP_FILENAME` to dump profile stats to a file, so they can be further evaluated, e.g. with `flameprof`.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -4,6 +4,7 @@ envlist = py27,py36,py37,py38,stylecheck
 [testenv]
 deps =
     pytest
+    flask<2.3.0 # For "before_first_request"
```
Comment (jeffwidman) [inline tox.ini:7]: Can you further clarify why this is needed?

We can't be pinning Flask to an old version unless it's a super temp pin until Flask drops a bugfix release... but Flask is now on `3.0.0` so methinks we'd want to test with that...

**Review by jeffwidman — COMMENTED** [review]
👍 for the general idea.

Can you rebase / squash to a single commit on top of `master`?

**Comment by Dosenpfand** [comment]
> Can you rebase / squash to a single commit on top of `master`?

Done.

File: tox.ini (hunk 1 above)
Comment (Dosenpfand) [reply tox.ini]: `test/basic_app.py` still uses the `before_first_request` decorator which was [removed in Flask 2.3.0](https://flask.palletsprojects.com/en/2.3.x/changes/#version-2-3-0).
Additional changes would be necessary to make the extension compatible with `3.0.0`. If desired, I can prepare an additional PR for that.

**Review by macnewbold — APPROVED** [review]
After the updates that @jeffwidman suggested, I think this looks good. Thanks for submitting it!

File: tox.ini (hunk 1 above)
Comment (jeffwidman) [reply tox.ini]: This is no longer part of the PR now that a few other PR's were merged re: compatibility.

File: src/flask_debugtoolbar/panels/profiler.py [hunk 2]
```diff
@@ -88,7 +92,14 @@ def process_response(self, request, response):
 
             self.stats = stats
             self.function_calls = function_calls
-            # destroy the profiler just in case
```
Comment (jeffwidman) [inline src/flask_debugtoolbar/panels/profiler.py:91]: Hmm, this comment seems semi-interesting, although I don't see an explicit `del` call here. Does it somehow jus t fall out of scope?

Just wondering if this is a contextual comment we should continue pulling forward in the codebase or if it's outdated.

**Comment by macnewbold** [comment]
I was able to resolve the merge conflict locally, but am running into permission errors when trying to push updates to the PR or to push updates that merge the PR into master. But if we can get past that, I think this one is ready to merge.

**Comment by jeffwidman** [comment]
@Dosenpfand can you rebase? And also address my last question?

File: src/flask_debugtoolbar/panels/profiler.py (hunk 2 above)
Comment (Dosenpfand) [reply src/flask_debugtoolbar/panels/profiler.py:91]: I thought it's outdated but if you think otherwise I can restore it.

**Comment by Dosenpfand** [comment]
@jeffwidman Done. Sorry for my late reaction.

**Comment by macnewbold** [comment]
It looks like it's all good except for one style check that is failing. Once that's fixed I think it is ready to merge.

**Comment by Dosenpfand** [comment]
@macnewbold 
fixed.

**Pull request merged**